File: atividade_5/java/src/Media.java
import java.util.ArrayList;
import java.util.Scanner;

public class Media {
    public static void main(String[] args) {
        try (var scan = new Scanner(System.in)) {
            var notasComPeso = new ArrayList<Double>();
            var pesos = new ArrayList<Double>();
            final int MARKS = 3;
            double nota = 0;
            double peso = 0;
            for (var i = 1; i <= MARKS; i++) {
                System.out.printf("Digite a %d° nota: ", i);
                nota = scan.nextDouble();
                System.out.printf("Digite o peso dessa nota: ");
                peso = scan.nextDouble();
                notasComPeso.add(nota * peso);
                pesos.add(peso);
            }
            var somaNotas = notasComPeso.stream().reduce(0d, Double::sum);
            var somaPesos = pesos.stream().reduce(0d, Double::sum);
            System.out.printf("A média Ponderada é: %.2f", somaNotas / somaPesos);
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }
}

File: atividade_5/C/media.c
#include <stdio.h>

int main(int argc, char const *argv[])
{
    double notas[3];
    int pesos[3];
    double nota = 0;
    int peso = 0;
    for (int i = 0; i < 3; i++)
    {
        printf("Digite a %u nota: \n", i+1);
        scanf("%lf", &nota);
        printf("Digite o peso dessa nota: \n");
        scanf("%u", &peso);
        notas[i] = nota * peso;
        pesos[i] = peso;
    }
    double somaNotas = 0;
    for (int i = 0; i < sizeof(notas) / sizeof(notas[0]); i++)
    {
        somaNotas += notas[i];
    }
    double somaPesos = 0;
    for (int i = 0; i < sizeof(pesos) / sizeof(pesos[0]); i++)
    {
        somaPesos += pesos[i];
    }
    printf("A media ponderada e: %.2f", somaNotas / somaPesos);
    return 0;
};